Demo/Common/IntegrationExamples/AdvancedFlowExample.swift

```diff
@@ -0,0 +1,374 @@
+        finish(with: error)
+    }
+
+}
+
+extension AdvancedFlowExample: PartialPaymentDelegate {
+
+    internal enum GiftCardError: Error, LocalizedError {
+        case noBalance
+
+        internal var errorDescription: String? {
+            switch self {
+            case .noBalance:
+                return "No Balance"
+            }
+        }
+    }
+
+    internal func checkBalance(with data: PaymentComponentData,
+                               component: Component,
+                               completion: @escaping (Result<Balance, Error>) -> Void) {
+        let request = BalanceCheckRequest(data: data)
+        apiClient.perform(request) { [weak self] result in
+            self?.handle(result: result, completion: completion)
+        }
+    }
+
+    private func handle(result: Result<BalanceCheckResponse, Error>,
+                        completion: @escaping (Result<Balance, Error>) -> Void) {
+        switch result {
+        case let .success(response):
+            handle(response: response, completion: completion)
+        case let .failure(error):
+            completion(.failure(error))
+        }
+    }
+
+    private func handle(response: BalanceCheckResponse, completion: @escaping (Result<Balance, Error>) -> Void) {
+        guard let availableAmount = response.balance else {
+            completion(.failure(GiftCardError.noBalance))
+            return
+        }
+        let balance = Balance(availableAmount: availableAmount, transactionLimit: response.transactionLimit)
+        completion(.success(balance))
+    }
+
+    internal func requestOrder(for component: Component,
+                               completion: @escaping (Result<PartialPaymentOrder, Error>) -> Void) {
+        let request = CreateOrderRequest(amount: ConfigurationConstants.current.amount,
+                                         reference: UUID().uuidString)
+        apiClient.perform(request) { [weak self] result in
+            self?.handle(result: result, completion: completion)
+        }
+    }
+
+    private func handle(result: Result<CreateOrderResponse, Error>,
+                        completion: @escaping (Result<PartialPaymentOrder, Error>) -> Void) {
+        switch result {
+        case let .success(response):
+            completion(.success(response.order))
+        case let .failure(error):
+            completion(.failure(error))
+        }
+    }
+
+    internal func cancelOrder(_ order: PartialPaymentOrder, component: Component) {
+        let request = CancelOrderRequest(order: order)
+        apiClient.perform(request) { [weak self] result in
+            self?.handle(result: result)
+        }
+    }
+
+    private func handle(result: Result<CancelOrderResponse, Error>) {
+        switch result {
+        case let .success(response):
+            if response.resultCode == .received {
+                presenter?.presentAlert(withTitle: "Order Cancelled", message: nil)
```

Comment:
It could nice to move content like this into an `enum`.

Reply:
This is a demo app and we want to keep it as simple as possible for merchant integration. So, I think for now we can keep it like this but in future if we feel like changing it we will surely do :)